src/cellfinder_core/detect/detect.py

```diff
@@ -37,26 +38,26 @@ def calculate_parameters_in_pixels(
 
 
 def main(
-    signal_array,
-    start_plane,
-    end_plane,
-    voxel_sizes,
-    soma_diameter,
-    max_cluster_size,
-    ball_xy_size,
-    ball_z_size,
-    ball_overlap_fraction,
-    soma_spread_factor,
-    n_free_cpus,
-    log_sigma_size,
-    n_sds_above_mean_thresh,
-    outlier_keep=False,
-    artifact_keep=False,
-    save_planes=False,
-    plane_directory=None,
+    signal_array: np.ndarray,
```

Comment:
I'm not sure this is always a numpy array (e.g. could be dask). Can we specify a numpy array like object? Or do we need do specify the union of several types?

Reply:
I've updated this to be a union of numpy and dask array.